api/overdrive.py

```diff
@@ -746,50 +749,23 @@ def json(self):
         return json.loads(self.content)
 
 
-class DummyOverdriveAPI(OverdriveAPI):
+class DummyOverdriveAPI(MockOverdriveAPI, OverdriveAPI):
```

Comment:
A lot of code in this class was a duplicate of code in core `MockOverdriveAPI`. Rather than change both classes to use the new `OverdriveAPI` technique, I just removed all the duplicate code. This had two main effects on the tests:

1. The circulation tests queue up mock responses by passing in Python dictionaries. The core tests queue up mock responses by passing in JSON strings. I had to change code in a number of places to treat a response body as either a JSON string or an object.

2. The circulation mock API incorrectly "queued" items by appending them to a list, instead of inserting them at the beginning of the list. The queue was actually a stack. This meant you had to "queue" responses from last to first. Now you queue responses in the same order as the mock HTTP requests are made.